### ISSUE TITLE
How to remove a control from a FormGroup?

### ISSUE THREAD

**Issue by naamapps** [joanpablo/reactive_forms#150]
Hey,
I'm trying to figure out how to remove a control from a FormGroup.
I didn't find any function like removeControl or something like that on the FormGroup object.

There is an addAll function which i'm using to add a control dynamically, and because it's dynamic, I also need to remove it dynamically.

Please help,
Thank you

**Comment by kuhnroyal** [comment]
Don't add controls dynamically, instead add them all at the start and disable those that you don't need. If you need to add groups, you can use a FormArray whichg supports adding and removing.

**Comment by naamapps** [comment]
@kuhnroyal I can't agree with you on this one.
It comes to code readability for this issue. 

I prefer to check that a control doesn't exist rather than checking if it's disabled.
Because it is actually doesn't need to be there.

If I check for disabled, than the next person that reads that code may think that the control is visible on the screen but disabled.
In my case it is not visible and it should be gone from the form group. 

Angular (which this package take a lot of inspiration from) has these two function
```
group.addControl()
group.removeControl()
```

So I really don't see any reason why this package shouldn't do the same.

**Comment by kuhnroyal** [comment]
I agree, this is just how I had to use it until now as I don't want to rebuild the whole form group every time.
Although this is mostly up to the use-case. You always have to know what you are doing, a disabled control can be readyOnly, disabled or actually removed in Flutter depending on your needs. It is the same for angular.

**Comment by joanpablo** [comment]
Hi @naamapps,

You are completely right when asking for a method for deleting a control from a group. I haven't included it yet because I haven't needed it before. I usually add features when I need them or other developers need them. 

Normally when I want to ignore a **value** (or pretend it does not exist) in a *Form*, I **disable** it, that way the value of the control will not be included in the value of the *FormGroup* and the validations will not take into account the disabled controls, so is like it does not exists in the Form. And later when I wanted to include it back I just enable it: **myControl.markAsEnabled()**.  For example:

```dart
// the email is disabled
final from = FormGroup({
   'name': FormControl<String>(value: 'John'),
   'email': FormControl<String>(value: 'johndoe@gmail.com', disabled: true)
});

print(form.value) // output: {'name': 'John'}

form.control('email').markAsEnabled();

print(form.value) // output: {'name': 'John', 'email': 'johndoe@gmail.com'}
```

That's the approach that I usually use, but of course, maybe in some cases somebody else could be wanted to **remove** completely the control from the Group, that's totally fine, and as you said it should be a reasonable behavior.

So please if you still need to completely remove the control, just let me know and I will include the functionality in the next release.

Thanks again for your support and Best regards.

**Comment by naamapps** [comment]
Hey @joanpablo, thanks for your reply.
It would be great if you could add it to the next release please. 
For now I will settle for the disabled solution, but in terms of code readability, it is not obvious that the control is not appearing on screen and its not actually disabled, but ignored. 

Thanks a lot!

**Comment by joanpablo** [comment]
I will include it in the next release. Thanks to all of you.

**Comment by joanpablo** [comment]
Hi @naamapps,

I have just released the version 10.3.0. In this version you can remove a control from a group by its name.

Please test it and if you find another issue don't hesitate to reopen this one or create a new one.

Best regards
